src/batch.js

```diff
@@ -0,0 +1,54 @@
+const call    = require('ramda/src/call')
+const curry   = require('ramda/src/curry')
+const juxt    = require('ramda/src/juxt')
+const max     = require('ramda/src/max')
+const zipWith = require('ramda/src/zipWith')
+
+// batch :: { k: v } -> ([a] -> Promise [b]) -> a -> Promise b
+const batch = (opts={}, f) => {
+  const {
+    limit = Infinity,
+    wait  = 32
+  } = opts
+
+  let args     = []
+  let last     = 0
+  let rejects  = []
+  let resolves = []
+  let timeout  = 0
+
+  const batched = arg => {
+    const promise = new Promise((res, rej) => {
+      args.push(arg)
+      rejects.push(rej)
+      resolves.push(res)
+    })
+
+    if (args.length >= limit) run()
+
+    else if (!timeout) {
+      const delta = new Date() - last
+      timeout = setTimeout(run, max(0, wait - delta))
+    }
+
+    return promise
+  }
+
+  const run = () => {
+    Promise.resolve(args)
```

Comment:
What happens when the `batched` function is called with the same arg more than once? Is there a good way to enforce that `args` is unique?

Reply:
It does not do any unique-ing.  The main driver for `batch` was to cut down on IO by batching requests from client to service, not to worry about whether duplicates will be inefficient for the service to serve.  My current use case is creating new records in a db, all of which will already be unique.  I could, however, build a map of args to promises, and use an existing promise if available before creating a new one.  Let me take a stab at that and push it up to see what you think.